### PR TITLE
fix(dashboard): correct mime type is returned. Fixes: #2290

### PR DESCRIPTION
**Pull request by nitram509**
## comments about this PR

I did refactor the code for static file serving into its own file.
Overall, I also did simplify the code and fixed former issues (see below).
Golang's mime package comes to the rescue when determining the correct mime type.

- add: tests for static files serving
- fix: when request /index.html, the base path was not modified
- fix: wrong 'err' variable used when log.Errorf("Failed to stat file or dir %s: %v"...) was printed

relates to #2290

PS: I would like to participate the hacktoberfest and would appreciated the 'hacktoberfest-accepted' label, IF you're OK with it :)
PPS: any feedback is welcome.

## Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by nitram509** [comment]
❓ I wonder 🤔  why the test-coverage went down?
The /server package did not contain any tests before, and I literally just created the first _test.go file in this package.
So I don't get, why the coverage is lower now since I wrote 4 additional test methods.

PS: I just measured the test coverage of the new file created, this is 87% (according to IntelliJ).

**Comment by nitram509** [comment]
⚠️ while further testing, I realized, the UI puts the namespace in the path ... that clashes with this implementation and I'm about to work on a fix

File: server/server_static.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	//go:embed static/*
+	static         embed.FS //nolint
+	staticBasePath = "static"
+	IndexHtmlFile  = staticBasePath + "/index.html"
```
Comment (zachaller) [inline server/server_static.go:21]: Is there a reason this is public?

Comment (nitram509) [reply server/server_static.go:21]: Oops, well spotted.
Not by Intend and can be private.

```suggestion
	indexHtmlFile  = staticBasePath + "/index.html"
```

**Review by zachaller — APPROVED** [review]

**Comment by zachaller** [comment]
Can you fix DCO I think your last commit missed it, then this will be good to merge thanks for contributing!

**Comment by nitram509** [comment]
Sure, will take care of it.

**Comment by nitram509** [comment]
fixed the DCO

**Review by zachaller — APPROVED** [review]

**Review by zachaller — APPROVED** [review]

**Pull request merged**